tests/test_device.py

```diff
@@ -1191,9 +1191,9 @@ def test_batch_transform_does_not_expand_supported_sum(self, mocker):
         """Tests that batch_transform does not expand Sums if they are supported."""
         H = qml.sum(qml.PauliX(0), qml.PauliY(0))
         qs = qml.tape.QuantumScript(measurements=[qml.expval(H)])
-        spy = mocker.spy(qml.transforms, "sum_expand")
+        spy = mocker.spy(qml.transforms, "split_non_commuting")
 
-        dev = self.SomeDevice()
+        dev = self.SomeDevice(shots=None)
```

Comment:
Why is this added?

Reply:
The original test was incorrect in this case, because the legacy device would only support `Sum` in analytic mode, not with finite shots. This is the case for `Hamiltonian`, so it should be the same for `Sum` and other multi-term observables too.

Reply:
I see! And when not providing shots, `SomeDevice` defaults to finite shots? :thinking:

Reply:
Yes, to 1000 shots.

Reply:
:o I was not aware of this default in the core library _at all_. Thanks!